### PR TITLE
Use std::string instead of QString in FunctionBrowser models and presenters

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR makes an effort to change the `QString`'s into `std::string`'s, and  `QStringList`'s into `std::vector`'s, which are used in the models and presenters related to the FunctionTree used in the Inelastic Data Analysis interface, the Muon ALC interface, and the Fit Script Generator interface. The reason for doing this is to ensure that we are conforming to the MVP pattern which requires for all Qt related code to only be inside the view. When debugging a problem, it is also difficult to read the QString and other Qt objects without a special plugin (which then causes `std::string`'s to be unreadable!). This is therefore a quality of life improvement for developers who need to debug issues in this area of the code.

This PR does not exhaustively change all `QString`'s and `QStringList`'s to `std::string`'s and `std::vector`'s. There are some instances such as in `FunctionBrowserUtils.cpp` where using `QString`'s provide a quick and convenient way of manipulating the data - so I have left these instances alone.

This needs to be merged before I can continue with #36325 

### To test:
I have already testing Inelastic Data Analysis. To test Fit Script Generator, follow the instructions on this page:
https://docs.mantidproject.org/nightly/interfaces/general/Fit%20Script%20Generator.html

To test ALC Muon interface:

**Part 1**
Set the default instrument to EMU, and turn on the data archive
2.Interfaces->Muon->ALC
For the Runs put `81061-81142`
Select field_danfysik for the log value
Tick the subtract box in the periods section and set the second combo box to 2.
In Calculation, set the max to 16
Click run and wait for the data to load

**Part 2**
Click Baseline Modelling
Add a linear background fit function
Add 2 fitting regions (by right clicking and adding sections)
Set one to be before the peak and one to be after the peak
Do the fit

**Part 3**
Go to the Peak Fitting
Add a Gaussian (put sigma = 20, centre = 2150 and height =0.02)
Do a fit

*This does not require release notes* because **there should be no change to user functionality**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/SingleFunctionTemplateBrowser.cpp [hunk 1]
```diff
@@ -56,9 +57,9 @@ void SingleFunctionTemplateBrowser::createProperties() {
   m_presenter.init();
 }
 
-void SingleFunctionTemplateBrowser::setDataType(const QStringList &allowedFunctionsList) {
+void SingleFunctionTemplateBrowser::setDataType(std::vector<std::string> const &allowedFunctionsList) {
   ScopedFalse _false(m_emitEnumChange);
-  m_enumManager->setEnumNames(m_fitType, allowedFunctionsList);
+  m_enumManager->setEnumNames(m_fitType, convertVectorToQStringList(allowedFunctionsList));
```
Comment (adriazalvarez) [inline qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/SingleFunctionTemplateBrowser.cpp:62]: In these 2 functions, is it necessary to add the prefix convert to the function name?
VectorToQStringList, or QStringListToVector is not ambiguous, right?

File: qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/SingleFunctionTemplateModel.h [hunk 2]
```diff
@@ -33,25 +33,25 @@ class MANTIDQT_INELASTIC_DLL SingleFunctionTemplateModel : public FunctionModel
...
 private:
-  QString m_fitType;
+  std::string m_fitType;
   DataForParameterEstimationCollection m_estimationData;
-  QMap<QString, IFunction_sptr> m_fitTypeToFunctionStore;
-  QMap<QString, QStringList> m_globalParameterStore;
-  QStringList m_fitTypeList;
-  boost::optional<QString> findFitTypeForFunctionName(const QString &name) const;
+  QMap<std::string, IFunction_sptr> m_fitTypeToFunctionStore;
+  QMap<std::string, std::vector<std::string>> m_globalParameterStore;
```
Comment (adriazalvarez) [inline qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/SingleFunctionTemplateModel.h:52]: some models still retain the QMap type, is there any advantage with respect to std::map, or it is something to perhaps do in a different pr?

**Review by adriazalvarez — CHANGES_REQUESTED** [review]
Thanks for the hard work in changing the QString dependencies. The changes are straightforward but the list is quite exhaustive!. Now models and presenters use the standard library strings, which will made code more robust and flexible to changes. I have tested it and everything works fine. 
I have couple of comments/questions, otherwise looks good to approve.

File: qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/SingleFunctionTemplateModel.h (hunk 2 above)
Comment (robertapplin) [reply qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/SingleFunctionTemplateModel.h:52]: I think ideally these should be `std::map` as well. In an attempt to keep this PR short (...I tried!) I decided to skip them. Are you happy if we change them in a future PR?

**Comment by robertapplin** [comment]
I am going to put this in draft until I get the Indirect-Inelastic dependency changes into `ornl-next`

**Comment by robertapplin** [comment]
I think we can go ahead with this

**Review by adriazalvarez — APPROVED** [review]
